libsolidity/formal/SMTEncoder.cpp

```diff
@@ -2582,12 +2586,12 @@ vector<VariableDeclaration const*> SMTEncoder::stateVariablesIncludingInheritedA
 	return stateVariablesIncludingInheritedAndPrivate(dynamic_cast<ContractDefinition const&>(*_function.scope()));
 }
 
-vector<VariableDeclaration const*> SMTEncoder::localVariablesIncludingModifiers(FunctionDefinition const& _function)
+vector<VariableDeclaration const*> SMTEncoder::localVariablesIncludingModifiers(FunctionDefinition const& _function, ContractDefinition const* _contract)
```

Comment:
The way this function is now called is
"give me the local variables of _function when in the context of _contract", right?
The difference from before is that _function might not be in _contract, and might have invocations to virtual/overridden modifiers which might be in _contract.
Correct?

Reply:
Yes, that's correct. The `_function` does not need to be in `_contract` directly, it can be that `_contract` just inherited it from one of its base contracts.
Exactly because `_contract` can have overriding modifiers used with `_function` and so the set of local variables can be different for different contracts in the inheritance hierarchy.

Reply:
Ok. Then I'm convinced it should be `contract` and not `base` in CHC

Reply:
OK, I think we converged on a decision and that's how it is done now in the code.